### PR TITLE
Minor refactoring for labOperator

### PR DESCRIPTION
**Pull request by palexster**
# Description

This PR refactors the creation of laboratory instances, by enabling two or more environments per template.

### CODE REVIEW

File: operators/pkg/laboratory-controller/labinstance_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+							Name:       "Test",
+							GuiEnabled: true,
+							Resources: crownlabsv1alpha2.EnvironmentResources{
+								CPU:                   1,
+								ReservedCPUPercentage: 1,
+								Memory:                resource.MustParse("1024M"),
+							},
+							EnvironmentType: crownlabsv1alpha2.ClassVM,
+							Persistent:      false,
+							Image:           "trololo/vm",
```
Comment (frisso) [inline operators/pkg/laboratory-controller/labinstance_controller_test.go:84]: trololo? :-)

**Comment by frisso** [comment]
First, great work!
Just two comments.
1) Not clear the object hierarchy to me. Templates, Environment, Instance; what is what? :-)
2) There are still some "lab" or "laboratory" in the code. Would it be replaced? "Lab" looks still the old name we used in the past; I would try to go beyond that.

File: operators/pkg/laboratory-controller/labinstance_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+/*
+The first step to writing a simple integration test is to actually create an instance of CronJob you can run tests against.
+Note that to create a CronJob, you’ll need to create a stub CronJob struct that contains your CronJob’s specifications.
+Note that when we create a stub CronJob, the CronJob also needs stubs of its required downstream objects.
+Without the stubbed Job template spec and the Pod template spec below, the Kubernetes API will not be able to
+create the CronJob.
+*/
```
Comment (giorio94) [inline operators/pkg/laboratory-controller/labinstance_controller_test.go:26]:
```suggestion
```

**Review by giorio94 — APPROVED** [review]
/lgtm

**Comment by palexster** [comment]
> First, great work!
> Just two comments.
> 
>     1. Not clear the object hierarchy to me. Templates, Environment, Instance; what is what? :-)
> 
>     2. There are still some "lab" or "laboratory" in the code. Would it be replaced? "Lab" looks still the old name we used in the past; I would try to go beyond that.

Totally agree. All those items will be addressed in the next refactoring PR.

**Review by frisso — APPROVED** [review]

**Pull request merged**